rust/relay/ebpf-turn-router/src/main.rs

```diff
@@ -107,54 +103,58 @@ pub fn handle_turn(ctx: XdpContext) -> u32 {
 
 #[inline(always)]
 fn try_handle_turn(ctx: &XdpContext) -> Result<u32, Error> {
-    // Safety: This is the only instance of `Eth`.
-    let eth = unsafe { Eth::parse(ctx) }?;
+    // SAFETY: This is the only instance of `EthHdr` in this scope.
+    let eth = unsafe { ref_mut_at::<EthHdr>(ctx, 0)? };
 
-    match eth.ether_type() {
-        EtherType::Ipv4 => try_handle_turn_ipv4(ctx, eth)?,
-        EtherType::Ipv6 => try_handle_turn_ipv6(ctx, eth)?,
+    match eth.ether_type {
+        EtherType::Ipv4 => try_handle_turn_ipv4(ctx)?,
+        EtherType::Ipv6 => try_handle_turn_ipv6(ctx)?,
         _ => return Err(Error::NotIp),
     };
 
     // If we get to here, we modified the packet and need to send it back out again.
-
     Ok(xdp_action::XDP_TX)
 }
 
 #[inline(always)]
-fn try_handle_turn_ipv4(ctx: &XdpContext, eth: Eth) -> Result<(), Error> {
-    // Safety: This is the only instance of `Ip4`.
-    let ipv4 = unsafe { Ip4::parse(ctx) }?;
+fn try_handle_turn_ipv4(ctx: &XdpContext) -> Result<(), Error> {
```

Comment:
Could we order the functions in the order they are used? I'd expect `try_handle_turn_ipv6` after this.